File: Javascript/decimal-to-binary.js

function decimalToBinary(dec){
    let binary='';
    while (dec > 0){
        let rem = dec % 2;
        binary = binary + rem;
        dec = Math.floor(dec/2);
    }
    return binary.split('').reverse().join('');
}

const res = decimalToBinary(32);
console.log(res);

File: ScalerTest/DecimalToBinary.java
public class DecimalToBinary {

    static long decimalToBinary(int n){
//        int p = 1;
        long ans = 0;
//        while (n > 0){
//            int rem = n % 2;
//            ans = ans + rem * p;
//            p = p * 10;
//            n = n / 2;
//        }
//        return ans;
        int p = 5;
        while (n > 0){
            int rem = n % 2;
            ans = ans + rem * p;
            p = p * 5;
            n = n / 2;
        }
        return ans;
    }
    public static void main(String[] args) {
        int a = 6;
        System.out.println(decimalToBinary(a));
    }
    /* rem = 0 + 0 * 5;
      ans =0;

    *
    * */
}
